packages/prosemirror-lwdita-backend/src/tests/github.controller.spec.ts

```diff
@@ -0,0 +1,190 @@
+
+describe('authenticateUserWithOctokit', () => {
+  let req: Partial<Request>;
+  let res: Partial<Response>;
+  let jsonStub: sinon.SinonStub;
+  let statusStub: sinon.SinonStub;
+
+  beforeEach(async () => {
+    // Mocking req and res objects
+    req = {
+      query: {}
+    };
+    jsonStub = sinon.stub();
+    statusStub = sinon.stub().returns({ json: jsonStub });
+    res = {
+      status: statusStub,
+      json: jsonStub
+    };
+  });
+
+  afterEach(() => {
+    sinon.restore(); // Restore original functionality
+  });
+
+  after(() => {
+  });
+
+  it('should return 400 if code is missing', async () => {
+    // Arrange: Leave req.query.code undefined
+
+    // Act: Call the controller
+    await authenticateUserWithOctokit(req as Request, res as Response);
+
+    // Assert: Check response status and message
+    expect(statusStub.calledWith(400)).to.be.true;
+    expect(jsonStub.calledWith({ message: 'Missing code' })).to.be.true;
+  });
+
+  it.skip('should return the token when code is provided', async () => {
+    // Arrange: Set req.query.code
+    if (!req.query) {
+      req.query = {};
+    }
+    req.query.code = 'test_code';
+    // const mockToken = { token: 'fake_token' };
+    // Mock the authenticateWithOAuth function
+    //FIXME(YB): Unable to mock authenticateWithOAuth function
+
+    // Act: Call the controller
+    await authenticateUserWithOctokit(req as Request, res as Response);
+
+    // Assert: Ensure the OAuth function was called and the correct response was returned
+    // expect(authenticateWithOAuthStub.calledWith('test_code')).to.be.true;
+    // expect(jsonStub.calledWith(mockToken)).to.be.true;
+  });
+});
+
+describe('getUserInformation', () => {
+  let req: Partial<Request>;
+  let res: Partial<Response>;
+  let jsonStub: sinon.SinonStub;
+  let statusStub: sinon.SinonStub;
+
+  beforeEach(async () => {
+    // Mocking req and res objects
+    req = {
+      query: {},
+      headers: {}
+    };
+    jsonStub = sinon.stub();
+    statusStub = sinon.stub().returns({ json: jsonStub });
+    res = {
+      status: statusStub,
+      json: jsonStub
+    };
+  });
+
+  afterEach(() => {
+    sinon.restore(); // Restore original functionality
+  });
+
+  it('should return 403 if authentication header is missing', async () => {
+    // Arrange: Leave req.headers.authorization undefined
+
+    // Act: Call the controller
+    await getUserInformation(req as Request, res as Response);
+
+    // Assert: Check response status and message
+    expect(statusStub.calledWith(403)).to.be.true;
+    expect(jsonStub.calledWith({ error: 'No credentials sent!' })).to.be.true;
+  });
+
+  it.skip('should return user information when token is provided', async () => {
+    // Arrange: Set req.headers.authorization
+    if (!req.headers) {
+      req.headers = {};
+    }
+    req.headers.authorization = 'Bearer test_token';
+
+    // Act: Call the controller
```

Comment:
Same as above

Reply:
So these comments: `Arrange` ,`ACT`, `Assert` are really connected to the stubbing methodology and I strongly recommend we should keep them in the tests.